server/core/src/miscServerFunct.cpp

```diff
@@ -2993,29 +2992,39 @@ irods::error get_script_output_single_line(
         if (!std::filesystem::exists(script_path)) {
             return ERROR(SYS_RESOLUTION_ERROR, fmt::format("Script file not found! {}", script_path));
         }
-        script_path_and_args.insert(
-            script_path_and_args.begin(),
-            fmt::format("{}/scripts/{}", irods::get_irods_home_directory().c_str(), script_name));
-        bp::ipstream program_output;
-        bp::child c(resolved_program, script_path_and_args, bp::std_out > program_output);
+        auto script_path_and_args = args;
+        script_path_and_args.insert(script_path_and_args.begin(), script_path);
+
+        bp::ipstream program_output, program_stderr;
+        bp::child c(resolved_program, script_path_and_args, bp::std_out > program_output, bp::std_err > program_stderr);
         output.clear();
         c.wait();
         const auto ec = c.exit_code();
+
         if (!program_output) {
-            return ERROR(ACTION_FAILED_ERR,
+            return ERROR(SYS_PIPE_ERROR,
                          fmt::format("stdout was unable to be opened, process exited with code {}", ec));
         }
         if (!std::getline(program_output, output)) {
-            return ERROR(ACTION_FAILED_ERR,
+            return ERROR(SYS_PIPE_ERROR,
                          fmt::format("Unable to read line from child process, exited with code {}", ec));
         }
+        std::string stderr_output;
+        if (program_stderr && std::getline(program_stderr, stderr_output) && !stderr_output.empty()) {
+            irods::experimental::log::server::error(
+                "Received stderr output '{}' from script process '{} {}' which exited with code",
```

Comment:
Missing fourth format placeholder.

Use square brackets around the placeholders. That is the normal iRODS convention. It helps with identifying whether the value was an empty string or not.